vllm/logger.py

```diff
@@ -1,61 +1,255 @@
+        }
+        return json.dumps(subset)
+
+    @classmethod
+    def formatter(cls, record):
+        """Format a log record for structured logging.
+
+        Note: This function returns the string to be formatted, not the actual
+        message to be logged.
+
+        Args:
+            record (dict): The log record to format.
+
+        Returns:
+            str: The formatted log record.
+        """
+        record["extra"]["serialized"] = cls.serialize(record)
+        return "{extra[serialized]}\n"
+
+    @classmethod
+    def customize_logging(
+        cls,
+        structured_filepath: Path = None,
+        unstructured_filepath: Path = None,
+        level: str = 'INFO',
+        rotation: str = None,
+        retention: str = None,
+        format: str = None,
+    ):
+        """Customize logging setup based on configuration options.
+
+        Configures logging with options for both structured and unstructured
+        log files, including rotation, retention, and formatting. Allows
+        enabling/disabling file logging and structured logging.
+
+        Args:
+            structured_filepath (Path): Path to the structured log file.
+            unstructured_filepath (Path): Path to the unstructured log file.
+            level (str): Logging level.
+            rotation (str): Log rotation policy.
+            retention (str): Log retention policy.
+            format (str): Log format.
+
+        Returns:
+            Logger: Configured logger instance.
+        """
+        logger.remove()
+
+        # Console output configuration
+        logger.add(
+            sys.stderr,
+            enqueue=True,
+            backtrace=True,
+            level=level.upper(),
+            format=format,
+        )
 
-class NewLineFormatter(logging.Formatter):
-    """Adds logging prefix to newlines to align multi-line messages."""
+        # File logging configuration
+        if unstructured_filepath:
+            # Unstructured file logging configuration
+            logger.add(
+                str(unstructured_filepath),
+                rotation=rotation,
+                retention=retention,
+                enqueue=True,
+                backtrace=True,
+                level=level.upper(),
+                serialize=False,
+                format=format,
+            )
+        if structured_filepath:
+            # Structured file logging configuration
+            logger.add(
+                str(structured_filepath),
+                rotation=rotation,
+                retention=retention,
+                enqueue=True,
+                backtrace=True,
+                level=level.upper(),
+                serialize=False,
+                format=cls.formatter,
+            )
 
-    def __init__(self, fmt, datefmt=None):
-        logging.Formatter.__init__(self, fmt, datefmt)
+        # Basic configuration for intercepting standard logging messages
+        logging.basicConfig(handlers=[InterceptHandler()], level=0)
```

Comment:
Does this screw up basic config from users? I think we should not touch the root config because vllm can be just imported and used as a library with other user code.

Reply:
Can we just setup the logger for "vllm" logger and emit to stdout from there instead?